src/app/backend/file-parser/path-parser.ts

```diff
@@ -40,45 +162,72 @@ export class PathParser {
                 }
             }
         }
+    }
+
+    parse(txt: string) {
+        // Verify structure of the input
+        this.preParse(txt);
+
+        // DOMParser for Control Points
+        const segmentCPs  = new DOMParser().parseFromString(txt, 'text/xml').getElementsByTagName('SegmentAppearance');
+        // DOMParse for Points Of Interest
+        const segmentPOIs = new DOMParser().parseFromString(txt, 'text/xml').getElementsByTagName('Segment');
+
+        // Used to keep track of all POI's and whether or not a POI has been fully initialized
+        const poiInitialized: boolean[] = [];
+        const pois: POI[] = [];
+        this.findAllPOIs(segmentPOIs, pois, poiInitialized);
 
+        const segments: Segment[] = [];
         // Iterate through each segment to find all control points
         // tslint:disable-next-line:prefer-for-of
         for (let i = 0; i < segmentCPs.length; i++) {
             // Get the Segment Entry of ControlPoints
-            const seg = segmentCPs[i];
+            const segCp = segmentCPs[i].children;
             // Get the Segment Entry of POIss
```

Comment:
POIss is used twice in this file and it appears to be a typign error (as I dont see what POIss would stand for), either be more clear on what POIss is or rename to POI's.
```suggestion
            // Get the Segment Entry of POI's
```

Reply:
See my other comment, shouldn't this become `POIs`?